src/Data/notifyOnConnectionTimeout.tsx

```diff
@@ -0,0 +1,54 @@
+import { currentLanguage } from 'scrivito'
+import { pseudoRandom32CharHex } from '../utils/pseudoRandom32CharHex'
+import { toast } from 'react-toastify'
+
+export async function notifyOnConnectionTimeout<T>(connection: Promise<T>) {
+  const connectionSucceeded = await Promise.race([
+    connection.then(() => true),
+    waitFor(5).then(() => false),
+  ])
+  if (connectionSucceeded) return
+
+  const toastId = pseudoRandom32CharHex()
```

Comment:
* [ ] As can be seen in https://fkhadra.github.io/react-toastify/update-toast#basic-example we get the ID as the result of the API call. No need to generate IDs -> simpler code.